content/courses/programmation/01-bases/05-boucles.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Les boucles"
+---
+
+En plus des conditions, il existe d'autres structures de contrôle très communes en programmation : les boucles.
+
+Nous savons désormais comment créer un tableau, mais si on devait écrire plusieurs fois la même logique pour chaque élément du tableau, notre programme serait un peu long. C'est le rôle d'une boucle, qui va appliquer le même bout de code à chaque élément d'une structure de données itérable.
+
+:::astride
+Pour les plus attentifs, on a déjà parlé des fonctions récursives qui permettent de faire des boucles. Mais dans cette section, nous allons voir d'autres outils très courants !
+:::
+
+### La boucle for in
+
+Imaginons que nous avons une liste d'ingrédients, et que nous voulons afficher dans la console chaque ingrédient. Je peux réaliser une boucle *for in* telle que :
+
+```ts
+let ingredients = list{"tomatoes", "cucumbers"}
+for (ingredient in ingredients) {
+    print(ingredient)
+}
+```
+
+Dans cette boucle, la variable `ingredient` va prendre tour à tour la valeur de chaque élément de la liste `ingredients`, et à chaque fois appliquer le bout de code qui se trouve à l'intérieur de la boucle.
```

Comment:
On est habitués que ce soit "singulier for pluriel" en tant que programmeur professionnel™ mais je pense que ça peut être un peu difficile à lire la première fois, peut-être juste écrire un petit truc "noter le singulier et le pluriel ici"